### PR TITLE
Add authentication

### PR DESCRIPTION
**Pull request by NipunaMarcus**
## Purpose
[Add authentication](https://github.com/LSFLK/MedicinesforLK/commit/d75bf41c9ab4893d2c4f2065e6f140460b0318fc)

### CODE REVIEW

File: admin-portal/src/components/navbar/navbar.tsx [hunk 1]
```diff
@@ -1,10 +1,66 @@
...
+  // const stateParam = new URLSearchParams(search).get("state");
+  // const errorDescParam = new URLSearchParams(search).get("error_description");
+
+  // useEffect(() => {
+  //   if (stateParam && errorDescParam) {
+  //     if (errorDescParam === "End User denied the logout request") {
+  //       setHasLogoutFailureError(true);
+  //     }
+  //   }
+  // }, [stateParam, errorDescParam]);
```
Comment (jayasanka-sack) [inline admin-portal/src/components/navbar/navbar.tsx:23]: Shall we remove this?

File: admin-portal/src/components/navbar/navbar.tsx [hunk 2]
```diff
@@ -1,10 +1,66 @@
...
+
+    //   const derivedState = {
+    //     authenticateResponse: basicUserInfo,
+    //     idToken: idToken?.split("."),
+    //     decodedIdTokenHeader: JSON.parse(atob(idToken?.split(".")[0])),
+    //     decodedIDTokenPayload: decodedIDToken
+    //   };
+
+    //   setDerivedAuthenticationState(derivedState);
+    // })();
```
Comment (jayasanka-sack) [inline admin-portal/src/components/navbar/navbar.tsx:52]: Do we need this later?

File: admin-portal/src/components/navbar/navbar.tsx [hunk 3]
```diff
@@ -1,10 +1,66 @@
...
+  }, [state?.isAuthenticated]);
+
+  const handleLogin = () => {
+    //setHasLogoutFailureError(false);
+    signIn(); //.catch(() => setHasAuthenticationErrors(true));
```
Comment (jayasanka-sack) [inline admin-portal/src/components/navbar/navbar.tsx:57]:
```suggestion
  const handleLogin = () => {
    signIn();
```

File: donor-portal/src/components/navbar/index.tsx [hunk 4]
```diff
@@ -1,31 +1,121 @@
...
+  // const stateParam = new URLSearchParams(search).get("state");
+  // const errorDescParam = new URLSearchParams(search).get("error_description");
+
+  // useEffect(() => {
+  //   if (stateParam && errorDescParam) {
+  //     if (errorDescParam === "End User denied the logout request") {
+  //       setHasLogoutFailureError(true);
+  //     }
+  //   }
+  // }, [stateParam, errorDescParam]);
```
Comment (jayasanka-sack) [inline donor-portal/src/components/navbar/index.tsx:35]:
```suggestion
```

**Review by jayasanka-sack — COMMENTED** [review]
Thanks for getting this done, @NipunaMarcus!
I added some comments. Could you please check them?
 
 Also, is there a specific reason to downgrade the react router?

File: admin-portal/src/components/navbar/navbar.tsx (hunk 2 above)
Comment (NipunaMarcus) [reply admin-portal/src/components/navbar/navbar.tsx:52]: yes we need this later to handle login errors.

File: admin-portal/src/components/navbar/navbar.tsx (hunk 3 above)
Comment (NipunaMarcus) [reply admin-portal/src/components/navbar/navbar.tsx:57]: Let's keep the commented code as we need these to handle errors

File: admin-portal/src/components/navbar/navbar.tsx (hunk 1 above)
Comment (NipunaMarcus) [reply admin-portal/src/components/navbar/navbar.tsx:23]: need these for the login error handling scenario

**Review by jayasanka-sack — APPROVED** [review]
Thanks @nipuna777 !

**Pull request merged**